python/src/ops.cpp

```diff
@@ -2931,20 +2931,41 @@ void init_ops(py::module_& m) {
       "load",
       &mlx_load_helper,
       "file"_a,
+      "format"_a = none,
       py::pos_only(),
       py::kw_only(),
       "stream"_a = none,
       R"pbdoc(
-        load(file: str, /, *, stream: Union[None, Stream, Device] = None) -> Union[array, Dict[str, array]]
+        load(file: str, format: Optional[str] = None, /, *, stream: Union[None, Stream, Device] = None) -> Union[array, Dict[str, array]]
 
         Load array(s) from a binary file in ``.npy`` or ``.npz`` format.
 
         Args:
             file (file, str): File in which the array is saved
-
+            format (str, optional): Format of the file. If ``None``, the format
+              is inferred from the file extension. Supported formats: npy, npz, and safetensors. (default: ``None``)
         Returns:
             result (array, dict): The loaded array if ``.npy`` file or a dict mapping name to array if ``.npz`` file
       )pbdoc");
+  m.def(
+      "save_safetensor",
+      &mlx_save_safetensor_helper,
+      "file"_a,
+      "d"_a,
```

Comment:
For consistency with `savez` should we do this as `args` (with default names) and `kwargs` - any reason not to follow that convention?

Reply:
I think the thinking behind the two are a little different, npz can be a collection of tensors but safetensors is thought of as a dictionary of key to tensor. So passing in a dict for safetensors feels more in line

Reply:
I'm cool with that.